crates/provider/src/provider/multicall/mod.rs

```diff
@@ -202,6 +202,23 @@ where
         self
     }
 
+    /// Add a dynamic call to the builder while specifying whether it is allowed to fail
+    ///
+    /// This is useful in combination with [`aggregate3`][MulticallBuilder::aggregate3].
+    pub fn add_dynamic3(
+        mut self,
+        item: impl MulticallItem<Decoder = D>,
+        allow_failure: bool,
+    ) -> Self {
+        let target = item.target();
+        let input = item.input();
+
+        let call = CallItem::<D>::new(target, input).allow_failure(allow_failure);
```

Comment:
these all operate on MulticallItem

should we make this an additional trait function

then we maybe don't this function if we can do

`item.into_call()` and `item.into_call_with(allow_failure)` or smth?

Reply:
I guess you mean `.add_call_dynamic(mycontract.someFunction(arg).into_call_with(true))` which would replace `add_dynamic3`

Reply:
Should I push the above conversions in the PR too?